src/sentry/api/endpoints/organization_api_key_details.py

```diff
@@ -0,0 +1,110 @@
+
+    def get(self, request, organization, api_key_id):
+        """
+        Retrieves API Key details
+        `````````````
+
+        :pparam string organization_slug: the slug of the organization the
+                                          team belongs to.
+        :pparam string api_key: the api key
+        :auth: required
+        """
+        try:
+            api_key = ApiKey.objects.get(
+                id=api_key_id,
+                organization_id=organization.id,
+            )
+        except ApiKey.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        return Response(serialize(api_key, request.user))
+
+    def put(self, request, organization, api_key_id):
+        """
+        Update an API Key
+        `````````````
+
+        :pparam string organization_slug: the slug of the organization the
+                                          team belongs to.
+        :pparam string api_key: the api key
+        :param string label: the new label for the api key
+        :param array scope_list: an array of scopes available for api key
+        :param string allowed_origins: list of allowed origins
+        :auth: required
+        """
+
+        try:
+            api_key = ApiKey.objects.get(
+                id=api_key_id,
+                organization_id=organization.id,
+            )
+        except ApiKey.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        serializer = ApiKeySerializer(api_key, data=request.DATA, partial=True)
+
+        if serializer.is_valid():
+            api_key = serializer.save()
+
+            self.create_audit_entry(
+                request=request,
+                organization=organization,
+                target_object=api_key_id,
+                event=AuditLogEntryEvent.APIKEY_EDIT,
+                data=api_key.get_audit_log_data(),
+            )
+
+            return Response(serialize(api_key, request.user))
+
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    def delete(self, request, organization, api_key_id):
+        """
+        Deletes an API Key
+        `````````````
+
+        :pparam string organization_slug: the slug of the organization the
+                                          team belongs to.
+        :pparam string api_key: the api key
+        :auth: required
+        """
+        try:
+            api_key = ApiKey.objects.get(
+                id=api_key_id,
+                organization_id=organization.id,
+            )
+        except ApiKey.DoesNotExist:
+            raise ResourceDoesNotExist
+
+        audit_data = api_key.get_audit_log_data()
+
+        api_key.delete()
+
+        self.create_audit_entry(
+            request,
+            organization=organization,
+            target_object=api_key.id,
+            event=AuditLogEntryEvent.APIKEY_REMOVE,
+            data=audit_data,
+        )
+
+        return Response(serialize(api_key, request.user), status=status.HTTP_202_ACCEPTED)
```

Comment:
is there a reason you're returning the serialized key? it looks like we generally just use `204` except in some cases where we use `202` when something has the possibility of failing asynchronously

Reply:
https://giphy.com/gifs/dexter-i-dont-know-what-im-doing-pushing-buttons-zLrMT1J1RuJ9u